Mathlib/NumberTheory/NumberField/Embeddings.lean

```diff
@@ -1005,6 +1007,143 @@ lemma card_eq_card_isUnramifiedIn [NumberField k] [IsGalois k K] :
       Finset.card (Finset.univ.filter <| IsUnramifiedIn K (k := k))ᶜ * (finrank k K / 2) := by
   rw [← card_isUnramified, ← card_isUnramified_compl, Finset.card_add_card_compl]
 
+variable {K : Type*} [Field K] [NumberField K] (v : InfinitePlace K)
+
+noncomputable instance : Inhabited (InfinitePlace K) :=
+  ⟨Classical.choice (instNonemptyInfinitePlaceOfNumberField K)⟩
+
+/-- The normed field structure of a number field coming from the norm asociated to
+an infinite place. -/
+noncomputable def normedField : NormedField K :=
+  NormedField.induced _ _ v.embedding v.embedding.injective
+
+/-- The normed division ring structure of a number field coming from the norm associated
+to an infinite place. -/
+noncomputable def normedDivisionRing : NormedDivisionRing K :=
+  v.normedField.toNormedDivisionRing
+
+/-- The uniform structure of a number field at an infinite place. -/
+noncomputable def uniformSpace : UniformSpace K :=
+  v.normedField.toUniformSpace
+
+/-- The uniform additive group structure of a number field induced by an infinite place. -/
+theorem uniformAddGroup : @UniformAddGroup K v.uniformSpace _ :=
+  @SeminormedAddCommGroup.to_uniformAddGroup _ (
+    @NormedAddCommGroup.toSeminormedAddCommGroup _ (
+      @NonUnitalNormedRing.toNormedAddCommGroup _ (
+        @NormedRing.toNonUnitalNormedRing _ (
+          @NormedDivisionRing.toNormedRing _ v.normedDivisionRing))))
```

Comment:
Can this be
```suggestion
theorem uniformAddGroup : @UniformAddGroup K v.uniformSpace _ := by
  letI := v.normedDivisionRing
  infer_instance
```
?

Reply:
yes, that's much nicer than chasing instances, thanks!